[18.0][MIG] product_medical: Migration to 18.0

LucasTran380381 opened the pull request:
Changes:

- remove string redundant in fields
- remove migrations directory 
- change tree view to list view

Commit:
[ADD] product_medical: create

Commit:
[IMP] product_medical: Change in layout, add multiple docs

1. The layout for the old tab MDR (now called Conformity)
   has been changed to split the fields into two categories:
   conformity-related and documents-related.

2. The two fields storing the "Declaration of Conformity" and
   the "CE Certificate Medical class", that before stored an
   attachment, now allow to store several attachments.
       To make this change, a migration script has been
   provided, because the field names have changed to adhere
   to the naming conventions.

3. The string for the field ce_certificate_medical_class_ids
   has changed from 'CE Certificate Medical class' to 'CE Certificate'.
   The translation files have been updated.

didierdonze committed:
Translated using Weblate (German)

Currently translated at 68.3% (28 of 41 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_medical
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_medical/de/

claudiagn committed:
Added translation using Weblate (Spanish)

claudiagn committed:
Added translation using Weblate (Catalan)

Commit:
[IMP] product_medical: views for conformity categories

The attributes for the Conformity Configuration (Medical Class,
Medical Category, PPE Category, and In Vitro Diagnostics) had no
views associated to them, so the only way of accessing them was
through the entry they have in the form for product.template.

With this PR, tree views and their actions have been added, so
that they can be reused in other modules to accessed them
from custom menu-items.

Commit:
[FIX] product_medical: minor changes in labels

- Medical Class -> Medical Device Class
- Medical Category > Drug Category

dzungtran89 committed:
[IMP] product_medical: black, isort, prettier

dzungtran89 committed:
[MIG] product_medical: Migration to 14.0

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (52 of 52 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_medical
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_medical/ca/

mymage committed:
Added translation using Weblate (Italian)

psc committed:
Translated using Weblate (German)

Currently translated at 84.6% (44 of 52 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_medical
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_medical/de/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (52 of 52 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_medical
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_medical/es/

Commit:
[IMP] product_medical: pre-commit auto fixes

Commit:
[MIG] product_medical: Migration to 18.0

rousseldenis commented:
/ocabot migration product_medical

rousseldenis reviewed (approved):
Code review.

Even if some fields in product template should maybe include 'medical' in their names to avoid possible conflicts

LucasTran380381 commented:
> Code review.
> 
> Even if some fields in product template should maybe include 'medical' in their names to avoid possible conflicts

@rousseldenis could you explain possible conflict ?

sebalix reviewed (commented):
Getting back on my first review: there are two pre-commit commits, and the 2nd one is bumping the version number wrongly, then it's reverted in the migration commit afterwards. This has to be fixed.

Regarding a "medical" prefix for fields that could conflicts with other non-medical topics, this is a good improvement indeed. Can you handle this with the proper migration script? :pray:

LucasTran380381 commented:
> Getting back on my first review: there are two pre-commit commits, and the 2nd one is bumping the version number wrongly, then it's reverted in the migration commit afterwards. This has to be fixed.
> 
> Regarding a "medical" prefix for fields that could conflicts with other non-medical topics, this is a good improvement indeed. Can you handle this with the proper migration script? 🙏

@sebalix thanks for your suggestion. I added migrations script and fixed commit

simahawk commented on product_medical/models/product_template.py:
```diff
@@ -7,34 +7,36 @@
+    medical_ce_certificate_medical_class_ids = fields.Many2many(
         "ir.attachment",
-        relation="product_ce_certificate_medical_class_rel",
+        relation="product_medical_ce_certificate_medical_class_rel",
```
These changes of relation tables are not migrated AFAIS.

sebalix replied:
`product_medical_ce_certificate_class_rel` based on my previous comment

sebalix commented on product_medical/migrations/18.0.1.0.0/pre-migrate.py:
```diff
@@ -0,0 +1,42 @@
+        [
+            (
+                model_name,
+                model_name.replace(".", "_"),
```
This conversion is not always true, to get the name of the table, that would be better to use `env[model_name]._table`, but at the same time:
- using the registry in a pre script is sometimes not a good idea
- no issue here for `product.template` data model, that will work

So not blocking :man_shrugging:

sebalix commented on product_medical/models/product_template.py:
```diff
@@ -7,34 +7,36 @@
-    ce_certificate_medical_class_ids = fields.Many2many(
+    medical_doc_lot_related = fields.Boolean(string="Lot Related", default=False)
+    medical_doc_validity_date = fields.Date(string="Validity Date")
+    medical_ce_certificate_medical_class_ids = fields.Many2many(
```
We could name it `medical_ce_certificate_class_ids` I guess, removing the second `medical` occurrence.

sebalix commented on product_medical/views/product_template.xml:
```diff
@@ -19,22 +19,22 @@
                         <group name="conformity_left" string="Categories">
                             <field name="medical_class_id" />
                             <field name="medicine_category_id" />
-                            <field name="ppe_category_id" />
```
We should rename the `conformity` page to `medical_conformity` still to avoid conflicts with other modules

sebalix reviewed (approved):
Tested on runboat, LG.
Regarding migration script, I could not test it but if something doesn't work we'll fix it ASAP anyway.

sebalix commented:
/ocabot merge nobump
